pkg/cmd/cli/secrets/source.go

```diff
@@ -0,0 +1,198 @@
+package secrets
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	CreateSourceSecretRecommendedCommandName = "new-sourcecfg"
+
+	createSourceSecretLong = `
+Create a new source secret
+
+Source secrets are used to authenticate against SCM servers.
+
+When creating applications, you may have a SCM server that requires authentication. In order for the
+nodes to clone source code on your behalf, they have to have the credentials. You can provide this
+information by creating a sourcecfg secret and attaching it to your service account.
+
+If your source authentication method requires only username and password or token, add it by using:
+
+  $ %[1]s SECRET_NAME --username=USERNAME --password=PASSWORD
+
+If your source authentication method requires private ssh key, add it by using:
+
+  $ %[1]s SECRET_NAME --ssh-private-key=SSH_KEY
+
+If your source authentication method requires a custom CA certificate, add it by using:
+
+  $ %[1]s SECRET_NAME --ca-cert=CERTIFICATE
+
+If you do already have a .gitconfig file needed for authentication, you can create a gitconfig secret by using:
+
+  $ %[2]s SECRET_NAME path/to/.gitconfig`
```

Comment:
Path to .gitconfig is a flag, not an argument, correct? So
`  $ %[2]s SECRET_NAME --gitconfig=FILENAME`